web/impact/impact/v1/helpers/model_helper.py

```diff
@@ -17,6 +37,96 @@ def field_value(self, field):
             return result
         return getattr(self.subject, field, None)
 
+    def field_setter(self, field, value):
+        subject = self.subject
+        # The following lines would allow a helper to
+        # override the subject's setter.  We haven't
+        # needed this yet, so leaving this mechanism
+        # commented out.
+        # attr = getattr(self.__class__, field, None)
+        # if attr and attr.fset:
+        #     subject = self
+        setattr(subject, field, value)
+
+    def validate(self, field, value):
+        validator = self.VALIDATORS.get(field)
+        if validator:
+            value = validator(self, field, value)
+        return value
+
+    def save(self):
+        self.subject.save()
+
     @classmethod
     def all_objects(cls):
         return cls.MODEL.objects.all()
+
+
+def validate_boolean(helper, field, value):
+    result = value
+    if isinstance(result, str):
+        result = result.lower()
+        if result == 'true':
+            result = True
+        elif result == 'false':
+            result = False
+    if not isinstance(result, bool):
+        helper.errors.append(INVALID_BOOLEAN_ERROR.format(field=field,
+                                                          value=value))
+    return result
+
+
+def validate_string(helper, field, value):
+    if not isinstance(value, str):
+        helper.errors.append(INVALID_STRING_ERROR.format(field=field,
+                                                         value=value))
+    return value
+
+
+def validate_choices(helper, field, value, choices, translations={}):
+    validate_string(helper, field, value)
+    result = translations.get(value, value)
+    if value in choices or result in choices:
+        return result
+    if isinstance(result, str):
+        result = translations.get(result.lower(), result)
```

Comment:
Because value isn't necessarily a string and I only wanted to try the .lower() if nothing else worked.  This is needed for "gender" so that "Male" gets properly translated to "m".